imaging/helper_funcs.py

```diff
@@ -0,0 +1,144 @@
+                                                nfiles), end="")
+        img = fabio.open(os.path.join(tiff_dir, filename))
+        stack.append(np.flipud(img.data))
+
+    print()  # Print a newline to separate each load message
+
+    return np.array(stack)
+
+
+def export_tiff_stack(dataset, key, base_name, output_dir, x_len, y_len):
+    to_save = dataset[key]
+
+    num_bins = 1 if len(to_save.shape) == 1 else to_save.shape[0]
+    stack_data = np.reshape(to_save.values, (x_len, y_len, num_bins))
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    # Writing tiffs
+    for i in range(stack_data.shape[2]):
+        imsave(os.path.join(output_dir,
+                            '{:s}_{:04d}.tiff'.format(base_name, i)),
+               stack_data[:, :, i].astype(np.float32))
+    print('Saved {:s}_{:04d}.tiff stack.'.format(base_name, 0))
+
+    # Write out tofs as CSV
+
+    if num_bins == 1:
+        dataset_tof = dataset.coords["tof"].values
+        tof_vals = [dataset_tof[0], dataset_tof[-1]]
+    else:
+        tof_vals = to_save.coords["tof"].values
+
+    with open(os.path.join(output_dir, 'tof_of_tiff_{}.txt'.format(base_name)),
+              'w') as tofs:
+        writer = csv.writer(tofs, delimiter='\t')
+        writer.writerow(['tiff_bin_nr', 'tof'])
+        tofs = tof_vals
+        tof_data = list(zip(list(range(len(tofs))), tofs))
+        writer.writerows(tof_data)
+    print('Saved tof_of_tiff_{}.txt.'.format(base_name))
+
+
+def tiffs_to_variable(tiff_dir):
+    """
+    Loads all tiff images from the directory into a scipp Variable.
+    """
+    stack = _load_tiffs(tiff_dir)
+    data = stack.astype(np.float64).reshape(stack.shape[0],
+                                            stack.shape[1] * stack.shape[2])
+    return sc.Variable(["tof", "spectrum"],
+                       values=data, variances=data)
+
+
+def stitch(data_array, frame_parameters, frame_shifts, rebin_parameters):
+    """
+    Stitches the 5 different frames data.
+
+    It crops out each frame, then shifts it so that all frames align,
+    and then rebins to the operations bins used for all frames.
+    """
+    frames = []
+
+    rebin_params = sc.Variable(["tof"], values=np.arange(*rebin_parameters,
+                                                           dtype=np.float64))
+
+    for i, (slice_bins, shift_parameter) in enumerate(
+            zip(frame_parameters, frame_shifts)):
+        bins = sc.Variable(["tof"],
+                           values=np.arange(*slice_bins, dtype=np.float64))
+        # Rebins the whole data to crop it to frame bins
+        rebinned = sc.rebin(data_array, "tof", bins)
+        # Shift the frame backwards to make all frames overlap
+        rebinned.coords["tof"] += shift_parameter
+        # Rebin to overarching coordinates so that the frame coordinates align
+        rebinned = sc.rebin(rebinned, "tof", rebin_params)
+
+        frames.append(rebinned)
```

Comment:
:+1: using rebin to crop instead of taking a view is really wasteful so fully agree

However, I suspect this will be a lot easier to do when we have value based slicing? Since we could say `data_array[start_tof : end_tof]` (or equivalent syntax) without having to figure out which index we need